src/machine.ts

```diff
@@ -116,7 +119,25 @@ export const SyncMachine = Machine<SyncContext, SyncSchema>({
       ],
     },
     reorg: {
-      type: 'final',
+      invoke: {
+        id: 'invokeReorg',
+        src: (_context, _event) => async () => {
+          const response = await invokeReorg();
+
+          if (!response.success) {
+            logger.error(response);
+            throw new Error('Reorg failed');
+          }
+
+          return;
+        },
+        onDone: {
+          target: 'idle',
+        },
+        onError: {
+          target: 'failure',
+        },
+      }
```

Comment:
This comment is not related to the changes in this PR, they are good for me. I was just re-reading the code where you set the 'reorg' type for the machine because it was already handled before.

I was just thinking about this line here: https://github.com/HathorNetwork/hathor-wallet-service-sync_daemon/blob/master/src/utils.ts#L260

I think we should try to consider this a reorg if `fullNodeBestBlock.height > ourBestBlock.height` but maybe we could just continue sending error to the machine and we handle this manually (we could just give more information to the error handler, e.g. `fullNodeBestBlock` and `ourBestBlock` heights)

What do you think?

Reply:
I think we can just transition to `reorg` and try to handle it on the wallet service

Done in b1871dcece4ca1b0e6b056ebcec6c0c58a7740b8